client/landing/stepper/index.tsx

```diff
@@ -65,22 +65,31 @@ interface AppWindow extends Window {
 	BUILD_TARGET?: string;
 }
 
-window.AppBoot = async () => {
+const DEFAULT_FLOW = 'site-setup';
+
+const getFlowFromURL = () => {
+	const fromPath = matchPath( { path: '/setup/:flow/*' }, window.location.pathname )?.params?.flow;
 	// backward support the old Stepper URL structure (?flow=something)
-	const flowNameFromQueryParam = new URLSearchParams( window.location.search ).get( 'flow' );
-	if ( flowNameFromQueryParam && availableFlows[ flowNameFromQueryParam ] ) {
-		window.location.href = `/setup/${ flowNameFromQueryParam }`;
+	const fromQuery = new URLSearchParams( window.location.search ).get( 'flow' );
+	return fromPath || fromQuery;
+};
+
+window.AppBoot = async () => {
+	const flowName = getFlowFromURL();
+
+	if ( ! flowName ) {
+		// Stop the boot process if we can't determine the flow, reducing the number of edge cases
+		return ( window.location.href = `/setup/${ DEFAULT_FLOW }${ window.location.search }` );
 	}
 
+	flowName === IMPORT_HOSTED_SITE_FLOW && addHotJarScript();
```

Comment:
Small favour, can we please make this an `if`? It confused the hell out of me for a good minute.

Reply:
Sure!  2d3b51c0fc6b24ae722561003f79f96335db5918